homeassistant/components/niko_home_control/config_flow.py

```diff
@@ -0,0 +1,88 @@
+"""Config flow for the Niko home control integration."""
+
+from __future__ import annotations
+
+import ipaddress
+from typing import Any
+
+from nikohomecontrol import NikoHomeControlConnection
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+
+from .const import DEFAULT_IP, DEFAULT_PORT, DOMAIN
+from .errors import CannotConnect, InvalidHost, InvalidPort
+
+
+async def validate_input(hass: HomeAssistant, data: dict) -> dict[str, str | int]:
+    """Validate the user input allows us to connect."""
+    host = data[CONF_HOST]
+    port = data[CONF_PORT]
+
+    try:
+        ipaddress.ip_address(host)
+    except ValueError:
+        raise InvalidHost from None
+
+    if port < 0 or port > 65535:
+        raise InvalidPort
+
+    controller = NikoHomeControlConnection(host, port)
+
+    if not controller:
+        raise CannotConnect
+
+    return {"host": host, "port": port}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Niko Home Control."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self, import_info=None) -> None:
+        """Initialize the config flow."""
+        self._import_info: Any = import_info
+
+    async def async_step_user(self, user_input=None) -> ConfigFlowResult:
```

Comment:
Do we have something that unique identifies the device we connect to?

Reply:
There is no mac address returned in the device info, was trying to include it myself but got cross platform issues.
So it's just identified by the name of the platform and by the action id.

Reply:
The id comes from the device indeed and is just a simple number, so we try to make it unique using platform & id

So a light is ```"light.0"```

I think I read something about they don't need to be unique for all domains but just for the (in this case) Niko one?
If that isn't the case I'm going to need to add the domain also and add a migration flow I guess.

Reply:
No I mean, unique id for the config entry. So we connect with a hub of some sort, does that hub have like a serial number so the next time someone tries to setup the same hub again, we can see that the serial number already has been used so we should abort and not allow new setup.

Reply:
I'll check but if I member correctly it only has sw version